src/main/java/com/etendoerp/sonar/checks/UseStringBuilderInsteadOfConcat.java

```diff
@@ -0,0 +1,72 @@
+package com.etendoerp.sonar.checks;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.sonar.check.Rule;
+import org.sonar.check.RuleProperty;
+import org.sonar.java.model.ExpressionUtils;
+import org.sonar.java.model.LiteralUtils;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.tree.BinaryExpressionTree;
+import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.LiteralTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+import com.etendoerp.sonar.utils.IssueMessages;
+
+@Rule(key = "UseStringBuilderInsteadOfConcat")
+public class UseStringBuilderInsteadOfConcat extends IssuableSubscriptionVisitor {
+
+  public static final int STR_LEN_DEFAULT_VALUE = 19;
+  private static final int DEFAULT_VALUE = 5;
+  private final Set<Tree> visitedNodes = new HashSet<>();
+  private final List<Tree> concats = new ArrayList<>();
+  @RuleProperty(
+      key = "threshold",
+      defaultValue = "" + DEFAULT_VALUE,
+      description = "Number of concatenations that have to be made to trigger an issue")
+  protected int threshold = DEFAULT_VALUE;
+
+  @RuleProperty(
+      key = "str_length",
+      defaultValue = "" + STR_LEN_DEFAULT_VALUE,
```

Comment:
The empty concatenation, is correct? `"" +`

Reply:
It's being used to cast the `STR_LEN_DEFAULT_VALUE` constant, from its original `int` type to a `String` type. I did it as such because some other sonar rules I found do the conversion like so